mu/integration_tests/test_babydriver.py

```diff
@@ -0,0 +1,88 @@
+import time
+import unittest
+import sys
+sys.path.append('projects/baby_driver/scripts')
+# pylint: disable=wrong-import-position
+import repl_setup
+import gpio_set
+import i2c_write
+import spi_exchange
+import gpio_interrupts
+import gpio_get
+import can_util
+import adc_read
+from mu.integration_tests import int_test
+
+
+
+class TestBabyDriver(int_test.IntTest):
+    def setUp(self):
+        super().setUp()
+        self.babydriver = self.manager.start('baby_driver')
+        can_util.default_channel = "vcan0"
+        repl_setup.initialize_modules()
+
+    def test_adc_read(self):
+        time.sleep(1)
+        adc_read.adc_read(port='A', pin=6, raw=False)
+
+    def test_gpio_get(self):
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=False)
+        time.sleep(1)
+        assert gpio_get.gpio_get('A', 5) is False
+
+    def test_gpio_set(self):
+        time.sleep(1)
+        self.babydriver.set_gpio(port='A', pin=5, state=False)
+        time.sleep(1)
+        gpio_set.gpio_set('A', 5, True)
+        time.sleep(1)
+        assert self.babydriver.get_gpio(port='A', pin=5) is True
+
+    def test_gpio_interrupts(self): #IN PROGRESS
+        # Defining a Callback function
+        # pylint: disable=attribute-defined-outside-init
+        isCalled = False
+        def interrupt_callback(info):
+            self.isCalled = True
```

Comment:
`isCalled` and `self.isCalled` are referring to two different variables: `isCalled` is a local variable in `test_gpio_interrupts`, whereas `self.isCalled` is an attribute on `self`. So this is one reason the assert in this test is failing.

To fix it, one solution is to use `self.isCalled` everywhere, but I think a nicer solution with fewer attributes defined on `self` (and which pylint won't complain about) would be to use `isCalled` in both places and put a [`nonlocal`](https://www.w3schools.com/python/ref_keyword_nonlocal.asp) declaration in `interrupt_callback`:
```py
def interrupt_callback(info):
    nonlocal isCalled
    isCalled = True
```
`nonlocal` is a Python 3 keyword which acts like `global`, but tells the interpreter that the variable referred to is in the next highest enclosing scope (`test_gpio_interrupts` in this case). Then you can remove the pylint comment here!